woocommerce/automations
# Add minimum required WC & WP versions in various files

This PR automates updating the requirement for WordPress and WooCommerce version support. Before, we needed to manually update various places in the codebase whenever we bump minimum version support.

- `readme.txt`
- `woocommerce-gutenberg-products-block.php`
- `phpcs.xml`

# Testing
To test this, you will need to:
- **Fork** the `woocommerce-gutenberg-products-block` and then unfork it (Go to this url: https://support.github.com/contact?tags=rr-forks and type `unfork` into the subject line. Follow the steps from the virtual assistant and wait 5-10 mins for the request to be actioned).
- Enable issues on your new unforked repository (go to `Settings > General > Features > Issues`)
- Activate GH actions
- Give read and write permissions to the workflows: Settings > Actions > General > Workflow permissions
- Allow GH actions to create PRs: Settings > Actions > General
- Create a milestone (I used 9.9.0, but pick whatever you like)
- Clone your repository locally.
- Create a PR on your repository, in this PR you should change the following file: https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/trunk/.github/workflows/release-pull-request.yml#L27
- Change the highlighted line to read: `uses: woocommerce/automations@add/min-required-wc-wp-versions`
- Merge the PR
- Optionally create multiple PRs.
- Assign them all to the milestone you created earlier.
- Create a new branch, `release/9.9.0` or whatever milestone number you picked.
- Push this branch to the remote.
- Go to your repository on GitHub, you should see a new PR created for the release, and it should have updated WordPress and WooCommerce versions in `readme.txt`, `woocommerce-gutenberg-products-block.php`, & `phpcs.xml`. 🙌🏼 

## TODO

- [x] Some refactoring is needed: we are handling the `.txt` & `.PHP` files in the same manner. Creating a common function will make the code cleaner!
